source/graphql/helper.d

```diff
@@ -462,6 +474,30 @@ T jsonTo(T)(Json item) {
 		} catch(Exception c) {
 			throw new GQLDExecutionException(c.msg);
 		}
+	} else static if(is(T == struct)) {
+		import std.traits : FieldNameTuple;
+		enforce!GQLDExecutionException(item.type == Json.Type.object,
+			format("'%s' was supposed to be an object", item.toPrettyString()));
+
+
+		T ret;
+		static foreach(mem; FieldNameTuple!T) {{
+			enum uda = getUdaData!(T, mem);
+			static if(uda.ignore != Ignore.yes) {{
```

Comment:
Have a look at the usage of IgnoreForInput as well. There is sort of a StructToJson for Input and for Output